examples/storage_proof.rs

```diff
@@ -0,0 +1,69 @@
+use axiom_eth::{keccak::KeccakChip, EthChip, storage::EthStorageChip, Field, providers::get_block_storage_input};
+use clap::Parser;
+use halo2_base::{AssignedValue, Context, gates::RangeInstructions};
+use halo2_scaffold::scaffold::{cmd::Cli, run_eth};
+use serde::{Deserialize, Serialize};
+use ethers_core::{types::{H160, H256}};
+use ethers_providers::{Provider, Http};
+use std::str::FromStr;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct CircuitInput {
+    pub provider_url: String,
+    pub state_root: String,
+    pub block_number: u32,
+    pub addr: String,
+    pub slot: u64,
+    pub acct_pf_max_depth: usize,
+    pub storage_pf_max_depth: usize,
+}
+
+fn compute_storage_proof_at_slot<F: Field>(
+    ctx: &mut Context<F>,
+    eth_chip: &EthChip<F>,
+    keccak: &mut KeccakChip<F>,
+    input: CircuitInput,
+    _: &mut Vec<AssignedValue<F>>,
+) -> impl FnOnce(&mut Context<F>, &mut Context<F>, &EthChip<F>) + Clone {
+    let provider = Provider::<Http>::try_from(input.provider_url.as_str()).unwrap();
+    let slot = H256::from_low_u64_be(input.slot);
+    let address = H160::from_str(&input.addr).unwrap();
+
+    // To prove the inlcusion of a storage we first need to fetch an mpt proof for a specific `slot` and block at `block_number`.
+    // The mpt proof will be used to generate constraints that the value at a the specified storage `slot` is included in the block.
```

Comment:
The MPT proof constitutes witness values and is not used to generate constraints.